### PR TITLE
[sdk/python]: use pynacl for signature generation and verification

### PR DESCRIPTION
**Pull request by gimre-xymcity**
> Note: This is the default pull request template.

## What's the issue?

SLOOOW

## How have you changed the behavior?

changed SDKs Signer and Verifier to use pynacl

## How was this change tested?

All existing tests pass, test vectors pass as well.
(Lint is passing ;) )

Branch might have wrong base and might need transplant onto sdk staging branch (?)

### CODE REVIEW

File: sdk/python/symbolchain/nem/KeyPair.py [hunk 1]
```diff
@@ -24,7 +45,36 @@ def private_key(self):
...
+
+		# pylint: disable=invalid-name
+		# r = H(privHash[256:512] || data)
+		r = _generate_nonce(self._sk, message)
+
+		# R = r * base point
+		r = crypto_core_ed25519_scalar_reduce(r)
+		R = crypto_scalarmult_ed25519_base_noclamp(r)
+
+		# h = H(encodedR || public || data)
```
Comment (Jaguar0625) [inline sdk/python/symbolchain/nem/KeyPair.py:57]: hram =

File: sdk/python/symbolchain/nem/KeyPair.py [hunk 2]
```diff
@@ -24,7 +45,36 @@ def private_key(self):
...
+		# hash and clamp private key
+		a = bytearray(sha3.keccak_512(self._sk).digest()[:32])
+		a[0] &= 0xF8
+		a[31] &= 0x7F
+		a[31] |= 0x40
+		a = bytes(a)
+
+		# S = (r + h * a) mod L
+		ha = crypto_core_ed25519_scalar_mul(a, hram)
+		S = crypto_core_ed25519_scalar_add(r, ha)
```
Comment (Jaguar0625) [inline sdk/python/symbolchain/nem/KeyPair.py:74]: do you want encoded S check, like in cpp ?

File: sdk/python/symbolchain/nem/KeyPair.py [hunk 3]
```diff
@@ -34,10 +84,48 @@ def __init__(self, public_key):
...
+		encoded_r = signature.bytes[:32]
+		encoded_s = signature.bytes[32:]
+
+		if not self.is_canonical_s(encoded_s):
+			return False
+
+		if bytes(32) == self._pk:
 			return False
+
+		# h = H(encodedR || public || data)
```
Comment (Jaguar0625) [inline sdk/python/symbolchain/nem/KeyPair.py:109]: hram =

File: sdk/python/symbolchain/nem/KeyPair.py [hunk 4]
```diff
@@ -34,10 +84,48 @@ def __init__(self, public_key):
...
+		hashobj.update(encoded_r)
+		hashobj.update(self._pk)
+		hashobj.update(message)
+		hram = hashobj.digest()
+		hram = crypto_core_ed25519_scalar_reduce(hram)
+
+		# note: this check covers following:
+		#  * (y) coord is canonical (< 2^255 - 19)
+		#  * has small order - this is slightly stronger check,
+		#    we don't do that in the client as we verify sigs with pub keys that are known to belong to accounts
```
Comment (Jaguar0625) [inline sdk/python/symbolchain/nem/KeyPair.py:120]: could client generate small order key ?

Comment (gimre-xymcity) [reply sdk/python/symbolchain/nem/KeyPair.py:120]: (discussed on discord)
tl;dr, should not be an issue on server part and subgroup check should be covering this as well

**Review by Jaguar0625 — APPROVED** [review]
